server/odc-service/src/main/java/com/oceanbase/odc/service/onlineschemachange/OscService.java

```diff
@@ -102,6 +115,17 @@ public OscSwapTableVO swapTable(@PathVariable Long scheduleTaskId) {
         OnlineSchemaChangeParameters oscParameters = JsonUtils.fromJson(scheduleEntity.get().getJobParametersJson(),
                 OnlineSchemaChangeParameters.class);
 
+        // check permission, only creator can swap table manual
+        PreConditions.validHasPermission(
+                Objects.equals(authenticationFacade.currentUserId(), scheduleEntity.get().getCreatorId()),
+                ErrorCodes.AccessDenied,
+                "no permission swap table name, schedule task id " + scheduleTaskId);
+
+        Optional<FlowInstance> optional = flowFactory.getFlowInstance(oscParameters.getFlowInstanceId());
```

Comment:
The `scheduleEntity` has a `organizationId` field and seems no need `Optional<FlowInstance>`

Reply:
scheduleEntity is not implements OrganizationIsolated

Reply:
ScheduleEntity is a entity object, so it dose not implements OrganizationIsolated. But the field ogranizationId is valid and could be used.
Anyway, your current code could work as you want.